background/services/chain/index.ts

```
@@ -238,34 +238,37 @@ export default class ChainService extends BaseService<Events> {
     // get the latest blocks and subscribe for all support networks
     // TODO revisit whether we actually want to subscribe to new heads
     // if a user isn't tracking a relevant addressOnNetwork
+    // eslint-disable-next-line no-restricted-syntax
     this.supportedNetworks.forEach(async (network) => {
       const provider = this.providerForNetwork(network)
       if (provider) {
-        const promises = Promise.all([
+        Promise.all([
           provider.getBlockNumber().then(async (n) => {
             const result = await provider.getBlock(n)
             const block = blockFromEthersBlock(network, result)
             await this.db.addBlock(block)
           }),
 
           this.subscribeToNewHeads(network),
-        ])
-        if (network.chainID !== ETHEREUM.chainID) {
-          // only block start to get Ethereum data
-          await promises
-        }
+        ]).catch((e) => {
+          logger.error("Error getting block number or new head", e)
+        })
       } else {
         logger.error(`Couldn't find provider for supported network ${network}`)
       }
     })
 
-    Promise.all(
+    Promise.allSettled(
```

This doesn't make a difference either way, right? I would expect if we use `allSettled` that we follow up with a `forEach` that handles the logging for any failed promises, rather than individually logging in each one.

---

I guess I wanted to use `allSettled` here to emphasize that we expect the possibility that some of the promises in the array fail rather than using `all` which implicitly rejects if even one of the promises fail.  And yeah in that context I definitely agree that a `forEach` to log errors is a win here.